app/src/main/java/me/hufman/androidautoidrive/music/spotify/SpotifyWebApi.kt

```diff
@@ -69,6 +69,33 @@ class SpotifyWebApi private constructor(val context: Context, val appSettings: M
 		clientId = SpotifyAppController.getClientId(context)
 	}
 
+	/**
+	 * Returns the songs from a specified playlist URI. This supports playlists that include both
+	 * songs and podcasts. Local tracks are not supported.
+	 */
+	suspend fun getPlaylistSongs(spotifyAppController: SpotifyAppController, playlistUri: String): List<SpotifyMusicMetadata>? = executeApiCall("Failed to get songs from playlist $playlistUri") {
+		if (webApi == null) {
+			return@executeApiCall emptyList()
+		}
+
+		val songs: ArrayList<SpotifyMusicMetadata> = ArrayList()
+		var pagedSongs = webApi?.playlists?.getPlaylistTracks(playlistUri, 50, 0, null)
+		while(pagedSongs != null) {
+			pagedSongs.items.forEach { playlistTrack ->
+				if (playlistTrack.track?.asTrack != null) {
```

Comment:
I think a more elegant approach is to try casting first and then use a null check:
```
val track = playlistTrack.track?.asTrack
if (track != null) {
  songs.add(createSpotifyMusicMetadataFromTrack(track, spotifyAppController))
}
```

Reply:
Ah I agree, updated the code with your suggested approach!